packages/beacon-state-transition/src/allForks/block/processDeposit.ts

```diff
@@ -75,7 +75,12 @@ export function processDeposit(fork: ForkName, state: CachedBeaconStateAllForks,
       slashed: false,
     });
     state.balanceList.push(Number(amount));
-    epochCtx.effectiveBalances.push(effectiveBalance);
+
+    // Updating here is better than updating at once on epoch transition
+    // - Simplify genesis fn applyDeposits(): effectiveBalanceIncrements is populated immediately
+    // - Keep related code together to reduce risk of breaking this cache
+    // - Should have equal performance since it sets a value in a flat array
+    epochCtx.effectiveBalanceIncrementsSet(validators.length, effectiveBalance);
```

Comment:
in average how many deposits we process per block? this create new `Uint8Array`, could we do this in batch?

Reply:
The effectiveBalanceIncrements Uint8Array is recreated in intervals of 1024 items. Once the length exceeds 1024 the length will be bumped to 2048.